src/Altinn.Platform/Altinn.Platform.Register/Register/Services/Implementation/PartiesWrapper.cs

```diff
@@ -21,30 +21,41 @@ public class PartiesWrapper : IParties
         private readonly GeneralSettings _generalSettings;
         private readonly ILogger _logger;
         private readonly HttpClient _client;
+        private readonly IMemoryCache _parties;
+        private const int _cacheTimeout = 5;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="PartiesWrapper"/> class
         /// </summary>
         /// <param name="httpClient">HttpClient from default httpclientfactory</param>
         /// <param name="generalSettings">the general settings</param>
         /// <param name="logger">the logger</param>
-        public PartiesWrapper(HttpClient httpClient, IOptions<GeneralSettings> generalSettings, ILogger<PartiesWrapper> logger)
+        /// <param name="partiesCache">memory cache for parties</param>
+        public PartiesWrapper(HttpClient httpClient, IOptions<GeneralSettings> generalSettings, ILogger<PartiesWrapper> logger, IMemoryCache partiesCache)
         {
             _generalSettings = generalSettings.Value;
             _logger = logger;
             _client = httpClient;
+            _parties = partiesCache;
         }
 
         /// <inheritdoc />
         public async Task<Party> GetParty(int partyId)
         {
+            if (_parties.TryGetValue(partyId, out Party party))
+            {
+                return party;
+            }
+
             Uri endpointUrl = new Uri($"{_generalSettings.BridgeApiEndpoint}parties/{partyId}");
 
             HttpResponseMessage response = await _client.GetAsync(endpointUrl);
 
             if (response.StatusCode == System.Net.HttpStatusCode.OK)
             {
-                return await JsonSerializer.DeserializeAsync<Party>(await response.Content.ReadAsStreamAsync());
+                party = await JsonSerializer.DeserializeAsync<Party>(await response.Content.ReadAsStreamAsync());
+                _parties.Set(party.PartyId, party, new TimeSpan(0, _cacheTimeout, 0));
```

Comment:
I would like to see the key being a little bit more specific. The MemoryCache service is a singleton. All classes asking for a MemoryCache will get the same instance.
```suggestion
                _parties.Set($"PartyId:{party.PartyId}", party, new TimeSpan(0, _cacheTimeout, 0));
```

Reply:
Using the key you suggested